[enterprise-4.9] Custom certs for the oAuth server route

slaskawi opened the pull request:
https://issues.redhat.com/browse/AUTH-13
https://issues.redhat.com/browse/OSDOCS-2366

This Pull Request introduces documentation bits for the Custom hostname and certificate for the oAuth server. It documents https://github.com/openshift/cluster-authentication-operator/pull/430

The documentation has been created based on https://github.com/openshift/openshift-docs/blob/master/modules/customizing-the-web-console-URL.adoc﻿

stlaz commented on modules/oauth-customizing-the-oauth-server-URL.adoc:
```diff
@@ -0,0 +1,36 @@
+        name: <secret-name>
+----
++
+If the domain for the custom hostname suffix does not match the cluster domain suffix, then a secret that contains a TLS certificate and key must exist in the `openshift-config` namespace and must be referenced in the `ingress` config. If the domain for the custom hostname suffix matches the cluster domain suffix, then the secret is optional. The secret must contain `tls.crt` and `tls.key` data k ...
```
I think English does not put a comma in front of "then" in conditionals.

awgreene replied:
It may be worth mentioning that there is a [TLS Secret](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets) designed for exactly this purpose.

slaskawi replied:
Removed comma.

Mentioned the TLS Secret.

bergerhoffer commented:
@slaskawi Thanks a lot for opening this! We haven't branched for 4.9 content yet, so I'm going to mark this WIP for now. But once we do, I'll come back and can take over any further editorial updates.

awgreene commented on modules/oauth-customizing-the-oauth-server-URL.adoc:
```diff
@@ -0,0 +1,36 @@
+
+.Prerequisites
+
+* Log in to the cluster as a user with administrative privileges.
```
Technically you just need to be able to update the spec of the `ingress.config.openshift.io/cluster` resource, but this is probably fine for documentation purposes.

slaskawi replied:
Ok, so leaving it as is.

awgreene reviewed (changes requested):
Great work @slaskawi! We may wish to mention how to review the status of the configured route as well.

slaskawi commented on modules/oauth-customizing-the-oauth-server-URL.adoc:
```diff
@@ -0,0 +1,36 @@
+// * authentication/configuring-internal-oauth.adoc
+
+[id="customizing-the-oauth-server-url_{context}"]
+= Customizing oAuth server URL
```
Good catch!

slaskawi commented on modules/oauth-customizing-the-oauth-server-URL.adoc:
```diff
@@ -0,0 +1,36 @@
+[id="customizing-the-oauth-server-url_{context}"]
+= Customizing oAuth server URL
+
+Custom route functionality for the oAuth server uses the `ingress` config route configuration API. A custom hostname and a TLS certificate can be set using the `spec.componentRoutes` part of the configuration.
```
Done

slaskawi commented on modules/oauth-customizing-the-oauth-server-URL.adoc:
```diff
@@ -0,0 +1,36 @@
+Custom route functionality for the oAuth server uses the `ingress` config route configuration API. A custom hostname and a TLS certificate can be set using the `spec.componentRoutes` part of the configuration.
+
+[id="customizing-the-oauth-server-route_{context}"]
+== Customizing the oAuth server route
```
Done

slaskawi commented on modules/oauth-customizing-the-oauth-server-URL.adoc:
```diff
@@ -0,0 +1,36 @@
+
+.Procedure
+
+* Set the custom hostname and, if needed, set TLS on the ingress config's spec:
```
Done

slaskawi commented:
Thanks for the review! I addressed all the comments and rebased it. 

> Thanks a lot for opening this! We haven't branched for 4.9 content yet, so I'm going to mark this WIP for now. But once we do, I'll come back and can take over any further editorial updates.

@bergerhoffer Sure, no problem! Thanks a lot!

s-urbaniak commented:
ping @bergerhoffer any chance to merge this?

bergerhoffer commented:
> ping @bergerhoffer any chance to merge this?

@s-urbaniak We just branched for 4.9 today. I'll go ahead and review this now, and if any updates are needed I can take care of those. Then we'll need QE review, but hopefully shouldn't be too long before we can get this merged.

bergerhoffer commented:
Closing this in favor of #34947. I'll tag there for further reviews.
